tests/test_autonomy/test_cli/test_helpers/test_fsm_spec.py

```diff
@@ -0,0 +1,170 @@
+
+
+def test_update_one() -> None:
+    """Test update_one"""
+
+    package_path = Path(hello_world_abci.__file__).parent.relative_to(ROOT_DIR)
+    with mock.patch.object(FSMSpecificationLoader, "dump") as m:
+        update_one(package_path)
+        m.assert_called_once()
+
+
+def test_update_one_raises() -> None:
+    """Test update_one raises"""
+
+    package_path = Path(packages.__file__).parent
+    expected = "FSM specification file .* does not exist, please provide app class name to continue."
+    with pytest.raises(ClickException, match=expected):
+        update_one(package_path)
+
+    package_path = Path(hello_world_abci.__file__).parent.relative_to(ROOT_DIR)
+    expected = "Please provide name for the app class or make sure FSM specification file is properly defined."
+    with mock.patch.object(FSMSpecificationLoader, "load", return_value={}):
+        with pytest.raises(ValueError, match=expected):
+            update_one(package_path)
+
+
+def test_check_one() -> None:
+    """Test check_one"""
+
+    package_path = Path(hello_world_abci.__file__).parent.relative_to(ROOT_DIR)
+    check_one(package_path)
+
+
+def test_check_one_raises() -> None:
+    """Test check_one raises"""
+
+    package_path = Path(hello_world_abci.__file__).parent.relative_to(ROOT_DIR)
+
+    expected = "Please provide name for the app class or make sure FSM specification file is properly defined."
+    with mock.patch.object(FSMSpecificationLoader, "load", return_value={}):
+        with pytest.raises(ValueError, match=expected):
+            update_one(package_path)
+
+    expected = 'Class .* is not in "packages.valory.skills.hello_world_abci.rounds"'
+    with pytest.raises(ClickException, match=expected):
+        check_one(package_path, app_class="DummyAbciApp")
+
+    target = "autonomy.cli.helpers.fsm_spec.check_unreferenced_events"
+    with mock.patch(target, return_value=["Event.WIN_LOTTERY"]):
+        expected = "Event reference check failed with .*"
+        with pytest.raises(DFASpecificationError, match=expected):
+            check_one(package_path)
+
+    with mock.patch.object(DFA, "load", return_value=""):
+        expected = "FSM Spec definition does not match in specification file and class definitions."
+        with pytest.raises(DFASpecificationError, match=expected):
+            check_one(package_path)
+
+
+@pytest.mark.parametrize("relative_path", [True, False])
+def test_check_all(relative_path: bool, capsys: CaptureFixture) -> None:
+    """Test check_all"""
+
+    packages_dir = Path(packages.__file__).parent
+    if relative_path:
+        packages_dir = packages_dir.relative_to(ROOT_DIR)
+
+    # there currently exists several fsm_specification.yaml files
+    with mock.patch("autonomy.cli.helpers.fsm_spec.check_one") as m:
+        check_all(packages_dir, FSMSpecificationLoader.OutputFormats.YAML)
+        m.assert_called()
+        captured = capsys.readouterr()
+        assert captured.out.count("Checking") > 3
```

Comment:
currently have 6 `fsm_specification.yaml` `>3` seems reasonable.

Reply:
Update this test to check the exact number rather than `>3`.

Reply:
but it will vary over time. Then this tests need unnecessarily be updated. I will check it if occurs (once) then